SSR-3283: build command update logic

hoangtran2506 opened the pull request:
## Why

There were some scripts not suitable for unit test implementation and need modification.


## What

Update build command logic for unit test.

## Checklist

- [x] Read [README.md](https://github.com/kintone-labs/kintone-cli/blob/master/README.md)
- [x] Updated documentation if it is required.
- [x] Checked the behavior.

TomokoMiyake commented on src/constant/error.ts:
```diff
@@ -1,14 +1,14 @@
+  VALID_DOMAIN: 'Please enter a valid domain.',
+  APP_NAME_MISSING: 'App name missing.',
   USER_NAME_EMPTY: "Username can't be empty.",
-  PASSWORD_EMPTY: "Password can't be empty",
+  PASSWORD_EMPTY: "Password can't be empty.",
   APP_ID_EMPTY: "App ID can't be empty.",
```
Please use a single quotation if it's ok.

hoangtran2506 replied:
In code
- We can't use this format: 'Password can't be empty.'
- We can use this format: 'Password can\'t be empty.', but prettier is not recommended.
- So I use "Password can't be empty."

TomokoMiyake replied:
I finally understood. It's due to the single quotation in "can't".
How about using "cannot" instead?

hoangtran2506 replied:
Sorry for the wrong description!!, I missing '\' in description above:

We can't use this format: 'Password can't be empty.'
We can use this format: 'Password can\'t be empty.', but prettier is not recommended.
So I use "Password can't be empty."

I updated code in here:
https://github.com/kintone-labs/kintone-cli/pull/118/commits/c47891cc978ff307674cac64d74f9357b6b4b769

TomokoMiyake replied:
I see!
Thanks for the modification. I confirmed.

TomokoMiyake commented on src/constant/error.ts:
```diff
@@ -1,14 +1,14 @@
+  PASSWORD_EMPTY: "Password can't be empty.",
   APP_ID_EMPTY: "App ID can't be empty.",
   APP_ID_NUMBER: 'App ID must be a number.',
   PROXY_EMPTY: "Proxy URL can't be empty."
```
Please use a single quotation if it's ok.

TomokoMiyake commented on dist/constant/message.js:
```diff
@@ -27,6 +27,6 @@ exports.MESSAGES = {
     WELCOME_MESSAGE: 'Welcome to kintone-cli!',
     WEBPACK_ENTRY_PROMPT: 'What is the entry for webpack?',
-    USE_PROXY_PROMT: 'Do you use proxy?'
+    USE_PROXY_PROMPT: 'Do you use proxy?'
```
It seems it's a duplication of line #23.
Please delete if it's ok.

hoangtran2506 replied:
This problem is resolved in here: https://github.com/kintone-labs/kintone-cli/pull/118#discussion_r1273285218

TomokoMiyake replied:
Got it.

TomokoMiyake commented on src/commands/Build/buildCommand.ts:
```diff
@@ -1,12 +1,11 @@
 
-const buildCommand = (program: Command) => {
-  program
+const buildCommand = (program: any) => {
+  return program
```
Due to this change, what will be changed with the behavior?

And for ts, it's not good to use "any".
Is it no problem with it?

hoangtran2506 replied:
The difference is that when call buildCommand() func, we can assign to variable to reuse program param in buildCommand func

I will change Command to commander.Command, because there's a bug when use Command:

![Screen Shot 2023-07-25 at 16 42 18](https://github.com/kintone-labs/kintone-cli/assets/117815925/042fc14c-2f37-4cb8-8b69-250a91ea1ef5)

hoangtran2506 replied:
And this bug make us can't run: npm run build

hoangtran2506 replied:
This is my resolve, Please have a check.
Thank you!
https://github.com/kintone-labs/kintone-cli/pull/118/files/8084b2d54c4c4942daa58c68591cef7144e2c7a5..1369b1a9b421b7aa4f371ae432bb11bcfa6804d1#diff-a4f691db069f66fae46d57dfa75429d03b45d4912e0825cb280fbb577d43a0ae

TomokoMiyake replied:
Got it. Thanks for the fix.

hoangtran2506 replied:
> Umm, it seems working fine with this. import { Command } from 'commander';

I use that import method, but it doesn't work: npm run build.
The reason in here:
https://github.com/tj/commander.js/issues/1465

TomokoMiyake replied:
I got it. Thanks for the reference, too.

hoangtran2506 replied:
Sorry for this inconvenient.
I used the wrong version of Commander (v8). 
I just updated the commander to new version (v11). It worked as expected.
I updated code in here:
https://github.com/kintone-labs/kintone-cli/pull/118/commits/b49db44d57bbaba871e4bb818bcd563185222560

Another command will not be affected

TomokoMiyake replied:
Thank you for the modification!

TomokoMiyake reviewed (changes requested):
When building the source code, there are some errors.
Is it ok not to change validator.ts?
And it's build command PR, but it includes authCommand change.
Please check it.

hoangtran2506 commented:
> When building the source code, there are some errors:
- I resolved in here: https://github.com/kintone-labs/kintone-cli/pull/118#discussion_r1273285218

>Is it ok not to change validator.ts: 
- It's ok in this time

>And it's build command PR, but it includes authCommand change. Please check it.
- I change Command to commander.Command, so it will have some difference

-

TomokoMiyake commented:
These files are added. Is it intended?
__tests__/test-helpers/buildOptionsTemp/
__tests__/test-helpers/buildTypeTemp/

TomokoMiyake commented:
> I change Command to commander.Command, so it will have some difference

I think you can update for other command source code in other PRs because it will be complex for reviewing.

hoangtran2506 commented:
> These files are added. Is it intended?

These files are added for unit test, it not effect to source code

> I think you can update for other command source code in other PRs because it will be complex for reviewing.
If I do that, we can't run: npm run build, because it will have a problem in other command files, like dev, Lint, Deploy, Auth, Initialize.
![Screen Shot 2023-07-25 at 16 42 18](https://github.com/kintone-labs/kintone-cli/assets/117815925/814861de-0614-465c-8405-9cd5e615f608)

TomokoMiyake commented:
> > These files are added. Is it intended?
> 
> These files are added for unit test, it not effect to source code
> 
> > I think you can update for other command source code in other PRs because it will be complex for reviewing.
> > If I do that, we can't run: npm run build, because it will have a problem in other command files, like dev, Lint, Deploy, Auth, Initialize.
> > ![Screen Shot 2023-07-25 at 16 42 18](https://user-images.githubusercontent.com/117815925/256084936-814861de-0614-465c-8405-9cd5e615f608.png)

I got it! If so, it's ok.

TomokoMiyake commented:
> > I change Command to commander.Command, so it will have some difference
> 
> I think you can update for other command source code in other PRs because it will be complex for reviewing.

How about this?

hoangtran2506 commented:
> > > I change Command to commander.Command, so it will have some difference
> > 
> > 
> > I think you can update for other command source code in other PRs because it will be complex for reviewing.
> 
> How about this?

I updated the resolve in here:
https://github.com/kintone-labs/kintone-cli/pull/118#discussion_r1274596490

TomokoMiyake reviewed (changes requested):
The auth command dist was also updated. I think it's not your intention.
Please build again with the latest source code.

hoangtran2506 commented:
> The auth command dist was also updated. I think it's not your intention. Please build again with the latest source code.

I updated build in here:
https://github.com/kintone-labs/kintone-cli/pull/118/commits/55ca7d3b8d49c8f840adb74cbcb51ebd38b25c49

TomokoMiyake commented:
> > The auth command dist was also updated. I think it's not your intention. Please build again with the latest source code.
> 
> I updated build in here: [55ca7d3](https://github.com/kintone-labs/kintone-cli/commit/55ca7d3b8d49c8f840adb74cbcb51ebd38b25c49)

Thank you!
Still there are auth command dist updates, but it's due to the revert and I think it's no problem with it.

TomokoMiyake reviewed (approved):
LGTM!
